src/System.IO.Pipes/src/System/IO/Pipes/PipeStream.Windows.cs

```diff
@@ -425,11 +425,14 @@ internal static unsafe Interop.Kernel32.SECURITY_ATTRIBUTES GetSecAttrs(HandleIn
                     secAttrs.bInheritHandle = Interop.BOOL.TRUE;
                 }
 
-                byte[] securityDescriptor = pipeSecurity.GetSecurityDescriptorBinaryForm();
-                pinningHandle = GCHandle.Alloc(securityDescriptor, GCHandleType.Pinned);
-                fixed (byte* pSecurityDescriptor = securityDescriptor)
+                if (pipeSecurity != null)
```

Comment:
Nitpick: it was already on the old code but you don't need the if statement on top, nor the new Interop.Kernel32.SECURITY_ATTRIBUTES(). Just use the one created on the top set its length (always) and let the other ifs do any of the extra work if needed.